clients/csrf/transport.go

```diff
@@ -17,38 +20,55 @@ type Cookies struct {
 }
 
 type Transport struct {
-	Transport http.RoundTripper
-	Csrf      *Csrf
-	Cookies   *Cookies
+	OriginalTransport http.RoundTripper
+	Csrf              *Csrf
+	Cookies           *Cookies
 }
 
 func (t Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	req2 := http.Request{}
 	copier.Copy(&req2, req)
 
-	if t.Cookies != nil {
-		UpdateCookiesIfNeeded(t.Cookies.Cookies, &req2)
-	}
-
-	csrfTokenManager := NewDefaultCsrfTokenUpdater(&t, &req2, NewDefaultCsrfTokenFetcher(&t))
+	UpdateCookiesIfNeeded(t.Cookies.Cookies, &req2)
 
-	err := csrfTokenManager.updateCsrfToken()
+	csrfTokenManager := NewDefaultCsrfTokenManager(&t, &req2)
+	err := csrfTokenManager.updateToken()
 	if err != nil {
 		return nil, err
 	}
 
-	res, err := t.Transport.RoundTrip(&req2)
+	log.Tracef("Sending a request with CSRF '" + req2.Header.Get("X-Csrf-Header") + " : " + req2.Header.Get("X-Csrf-Token") + "'\n")
+	log.Tracef("The sticky-session headers are: " + prettyPrintCookies(req2.Cookies()) + "\n")
+	res, err := t.OriginalTransport.RoundTrip(&req2)
 	if err != nil {
 		return nil, err
 	}
-	isRetryNeeded, err := csrfTokenManager.isRetryNeeded(res)
+	tokenWasRefreshed, err := csrfTokenManager.refreshTokenIfNeeded(res)
 	if err != nil {
 		return nil, err
 	}
 
-	if isRetryNeeded {
+	if tokenWasRefreshed {
+		log.Tracef("Response code '" + strconv.Itoa(res.StatusCode) + "' from bad token. Must Retry.\n")
 		return nil, &ForbiddenError{}
 	}
 
 	return res, err
 }
+
+func UpdateCookiesIfNeeded(cookies []*http.Cookie, request *http.Request) {
+	if cookies != nil && len(cookies) > 0 {
```

Comment:
`len()` is nil safe. You can delete the first condition. Also, woudn't it be better if the `if` was inverted, like so:
``` go
if len(cookies) == 0 {
   return
}
...
```